rust_src/src/obarray.rs

```diff
@@ -90,14 +91,14 @@ impl From<LispObject> for Option<LispObarrayRef> {
 }
 
```

Comment:
Is it time to switch to `LispSymbolRef`? It seems to just lead to lots of "turn it back into a LispObject" at the moment.

Reply:
Yeah, there's a lot of that, especially when it comes to interacting with C functions. On the other hand, we get rid of clutter like `let sym = symbol.as_symbol_or_error();`. We can continue to smooth it out over time.